packages/graph/hash_graph/lib/graph/src/store/postgres/query/database.rs

```diff
@@ -1,57 +1,44 @@
-use std::{borrow::Cow, fmt, fmt::Write};
+use std::{
+    borrow::Cow,
+    fmt::{self, Formatter, Write},
+};
 
 use serde::Serialize;
 
 use crate::store::postgres::query::Transpile;
 
-/// A table available in the database.
+/// An identifier of a [`Table`] in the data base
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Serialize)]
 #[serde(rename_all = "snake_case")]
-pub enum Table {
+pub enum TableName {
     TypeIds,
 }
 
-impl Table {
+impl TableName {
     /// Returns the [`Column`] used for joining another `Table` on this `Table`.
-    pub const fn source_join_column(self) -> Column<'static> {
-        let column = match self {
+    pub const fn source_join_column(self) -> &'static str {
+        match self {
             Self::TypeIds => "version_id",
-        };
-        Column {
-            table: self,
-            access: ColumnAccess::Table { column },
         }
     }
 
     /// Returns the [`Column`] used for joining this `Table` on another `Table`.
-    pub const fn target_join_column(self) -> Column<'static> {
-        let column = match self {
+    pub const fn target_join_column(self) -> &'static str {
+        match self {
             Self::TypeIds => "version_id",
-        };
-        Column {
-            table: self,
-            access: ColumnAccess::Table { column },
         }
     }
 }
 
-impl Transpile for Table {
-    fn transpile(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
-        fmt.write_char('"')?;
-        self.serialize(&mut *fmt)?;
-        fmt.write_char('"')
-    }
-}
-
-/// A table available in a compiled query.
+/// Alias parameters used to uniquely identify a [`Table`].
 ///
-/// When joining tables, each table requires a unique name. A `TableRef` may either be a plain
-/// [`Table`], or a [`Table`] with additional information attached to uniquely identify the table.
+/// When joining tables, each table requires a unique name. A [`Table`] may requires additional
+/// information attached to uniquely identify it.
```

Comment:
```suggestion
/// When joining tables in a query, it's necessary that the names used to reference them are unique. Achieving this can require aliasing the names if the various parts of the query rely on the same [`Table`] but under different conditions. To appropriately identify a [`Table`] when aliased, some additional information associated with it may be needed.
```

Maybe that? (Apologies about the line breaks as always)

Reply:
Sounds good, thanks!